python/lsst/ts/observatory/control/maintel/comcam.py

```diff
@@ -240,7 +241,43 @@ async def setup_filter(self, filter):
         if filter is None:
             return
         else:
-            raise NotImplementedError("set_filter not available yet.")
+            self.rem.cccamera.cmd_setFilter.set(name=filter)
+            async with self.cmd_lock:
+                self.rem.cccamera.evt_endSetFilter.flush()
+                await self.rem.cccamera.cmd_setFilter.start(
+                    timeout=self.filter_change_timeout
+                )
+                end_setFilter = await self.rem.cccamera.evt_endSetFilter.next(
+                    flush=False, timeout=self.filter_change_timeout
+                )
+                self.log.info(f"Filter {end_setFilter.filterName} in position.")
+                return end_setFilter
+
+    async def get_current_filter(self):
+        """Get the current filter.
+
+        Returns
+        -------
+        `str`
+            The filter in the light path.
+        """
+        end_setFilter = await self.rem.cccamera.evt_endSetFilter.aget(
+            timeout=self.fast_timeout
+        )
+        return end_setFilter.filterName
+
+    async def get_available_filters(self):
+        """Get the list of available filters.
+
+        Returns
+        -------
+        `str`
+            The set of filters available (colon delimited)
+        """
+        available_filters = await self.rem.cccamera.evt_availableFilters.aget(
+            timeout=self.fast_timeout
+        )
+        return available_filters.filterNames
```

Comment:
Do you want to split this string and return a list of strings?

Reply:
List of strings sounds better. Done.